File: src/lightnap-ng/src/app/core/backend-api/dtos/static-contents/request/search-static-content-request-dto.ts
import { PagedRequestDto, StaticContentReadAccess, StaticContentSortBy, StaticContentStatus, StaticContentType } from "@core/backend-api";

/**
 * Represents a request to search for static contents with specific criteria.
 */
export interface SearchStaticContentRequestDto extends PagedRequestDto {
  keyContains?: string;
  readAccess?: StaticContentReadAccess;
  status?: StaticContentStatus;
  type?: StaticContentType;
  sortBy: StaticContentSortBy;
  reverseSort: boolean;
}

File: src/LightNap.Core/StaticContents/Dto/Request/SearchStaticContentRequestDto.cs
using LightNap.Core.Api;
using LightNap.Core.StaticContents.Enums;
using LightNap.Core.StaticContents.Models;

namespace LightNap.Core.StaticContents.Dto.Request
{
    public class SearchStaticContentRequestDto : PagedRequestDtoBase
    {
        /// <summary>
        /// Substring key search filter.
        /// </summary>
        public string? KeyContains { get; set; }

        /// <summary>
        /// Filter for read access.
        /// </summary>
        public StaticContentReadAccess? ReadAccess { get; set; }

        /// <summary>
        /// Filter for status.
        /// </summary>
        public StaticContentStatus? Status { get; set; }

        /// <summary>
        /// Filter for type.
        /// </summary>
        public StaticContentType? Type { get; set; }

        /// <summary>
        /// Sorting order for the results.
        /// </summary>
        public StaticContentSortBy SortBy { get; set; }

        /// <summary>
        /// Reverses the default sort behavior of the SortBy field.
        /// </summary>
        public bool ReverseSort { get; set; }
    }
}